core/Archive/ArchivePurger.php

```diff
@@ -231,6 +279,27 @@ protected function getOutdatedArchiveIds(Date $date, $purgeArchivesOlderThan)
         return $idArchivesToDelete;
     }
 
+    /**
+     * returns the ids of archives which are in a broken state (missing the done flag)
+     * Only searches the archive linked to the date specified in $dateStart
+     *
+     * @param Month $month The month period describing which table to search for broken archives
+     * @return array
+     */
+    private function getBrokenArchiveIds(Month $month): array
+    {
+        $archiveTable = ArchiveTableCreator::getNumericTable($month->getDateStart());
+
+        $result = $this->model->getArchivesMissingDoneFlag($archiveTable);
+
+        $idArchivesToDelete = [];
+        if (!empty($result)) {
+            $idArchivesToDelete = array_column($result, 'idarchive');
+        }
+
+        return $idArchivesToDelete;
```

Comment:
we may omit the variable and use an early return
```suggestion
        if (!empty($result)) {
            return array_column($result, 'idarchive');
        }

        return [];
```

Reply:
done